From f3f87471b0a1ff3fd35b6cce44e5bb79abbe5d6a Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Wed, 18 Jul 2018 11:52:43 +0300
Subject: [PATCH 602/767] staging: fsl-mc/dpio: Endianness fixes

Introduce several fixes to the fq/bpool query functions:
- add endianness conversions where necessary
- fix an incorrect masking (mask needs to be applied before
endianness conversion)

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-mc/bus/dpio/qbman-portal.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/staging/fsl-mc/bus/dpio/qbman-portal.c b/drivers/staging/fsl-mc/bus/dpio/qbman-portal.c
index b4d0a5272649..5e31cd176b3c 100644
--- a/drivers/staging/fsl-mc/bus/dpio/qbman-portal.c
+++ b/drivers/staging/fsl-mc/bus/dpio/qbman-portal.c
@@ -1084,7 +1084,7 @@ int qbman_fq_query_state(struct qbman_swp *s, u32 fqid,
 		return -EBUSY;
 
 	/* FQID is a 24 bit value */
-	p->fqid = cpu_to_le32(fqid) & 0x00FFFFFF;
+	p->fqid = cpu_to_le32(fqid & 0x00FFFFFF);
 	resp = qbman_swp_mc_complete(s, p, QBMAN_FQ_QUERY_NP);
 	if (!resp) {
 		pr_err("qbman: Query FQID %d NP fields failed, no response\n",
@@ -1107,12 +1107,12 @@ int qbman_fq_query_state(struct qbman_swp *s, u32 fqid,
 
 u32 qbman_fq_state_frame_count(const struct qbman_fq_query_np_rslt *r)
 {
-	return (r->frm_cnt & 0x00FFFFFF);
+	return (le32_to_cpu(r->frm_cnt) & 0x00FFFFFF);
 }
 
 u32 qbman_fq_state_byte_count(const struct qbman_fq_query_np_rslt *r)
 {
-	return r->byte_cnt;
+	return le32_to_cpu(r->byte_cnt);
 }
 
 struct qbman_bp_query_desc {
@@ -1132,7 +1132,7 @@ int qbman_bp_query(struct qbman_swp *s, u32 bpid,
 	if (!p)
 		return -EBUSY;
 
-	p->bpid = bpid;
+	p->bpid = cpu_to_le16(bpid);
 	resp = qbman_swp_mc_complete(s, p, QBMAN_BP_QUERY);
 	if (!resp) {
 		pr_err("qbman: Query BPID %d fields failed, no response\n",
@@ -1155,5 +1155,5 @@ int qbman_bp_query(struct qbman_swp *s, u32 bpid,
 
 u32 qbman_bp_info_num_free_bufs(struct qbman_bp_query_rslt *a)
 {
-	return a->fill;
+	return le32_to_cpu(a->fill);
 }
-- 
2.17.0

